src/main/java/me/fromgate/reactions/logic/ItemStoragesManager.java

```diff
@@ -53,15 +53,32 @@ public void triggerItemHold(Player player) {
 
     private void setFutureItemHoldCheck(final UUID playerId, final String itemStr, boolean repeat) {
         Player player = Bukkit.getPlayer(playerId);
+
         if (player == null || !player.isOnline() || player.isDead()) return;
-        ItemStack item = player.getInventory().getItemInMainHand();
-        if (!ItemUtils.isExist(item)) return;
+
+        ItemStack mainHandItem = player.getInventory().getItemInMainHand();
+        ItemStack offHandItem = player.getInventory().getItemInOffHand();
+
+        boolean mainHandItemExist = isItemHoldProcessable(mainHandItem, itemStr);
+        boolean offHandItemExist = isItemHoldProcessable(offHandItem, itemStr);
+
+        if (!mainHandItemExist && !offHandItemExist) return;
         String rg = "ih-" + itemStr;
         if (!StoragesManager.isTimeToRaiseEvent(player, rg, Cfg.itemHoldRecheck, repeat)) return;
-        if (!ItemUtils.compareItemStr(item, itemStr)) return;
+
+        if (mainHandItemExist) processItemHoldActivator(player, mainHandItem);
+        if (offHandItemExist) processItemHoldActivator(player, offHandItem);
+
+        Bukkit.getScheduler().runTaskLater(ReActions.getPlugin(), () -> setFutureItemHoldCheck(playerId, itemStr, true), 20 * Cfg.itemHoldRecheck);
+    }
+
+    private void processItemHoldActivator(Player player, ItemStack item) {
         ItemHoldStorage ihe = new ItemHoldStorage(player, item, true);
```

Comment:
Consider implementing the hand option. For implementing references check out `ITEM_CLICK` [storage](https://github.com/imDaniX/ReActions/blob/2aa45e1ef859cc39ea3ba0c829705f07e3ce6724/src/main/java/me/fromgate/reactions/logic/storages/ItemClickStorage.java) and [activator](https://github.com/imDaniX/ReActions/blob/2aa45e1ef859cc39ea3ba0c829705f07e3ce6724/src/main/java/me/fromgate/reactions/logic/activators/ItemClickActivator.java) classes.

Reply:
```
private final boolean mainHand;
```
Storage uses boolean type for determining the mainHand though, should i replace it to HandType?

Reply:
Hand is checked by using this boolean in the [`HandType#checkMain(boolean isMain)`](https://github.com/imDaniX/ReActions/blob/2aa45e1ef859cc39ea3ba0c829705f07e3ce6724/src/main/java/me/fromgate/reactions/util/enums/HandType.java#L14) method. If you're going to replace it with the `HandType` class, it will definitely increase the readability, but then consider creating a new method `HandType#accepts(HandType hand)` and use the `@Deprecated` annotation on the other two.